packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/05-updating-versions.md

```diff
@@ -0,0 +1,357 @@
+
+/// @title SimpleStorageSetup build 2
+contract SimpleStorageBuild2Setup is PluginSetup {
+  address private immutable simpleStorageImplementation;
+
+  constructor() {
+    simpleStorageImplementation = address(new SimpleStorageBuild2());
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareInstallation(
+    address _dao,
+    bytes memory _data
+  ) external returns (address plugin, PreparedSetupData memory preparedSetupData) {
+    (uint256 _number, address _account) = abi.decode(_data, (uint256, address));
+
+    plugin = createERC1967Proxy(
+      simpleStorageImplementation,
+      abi.encodeWithSelector(SimpleStorageBuild2.initializeBuild2.selector, _dao, _number, _account)
+    );
+
+    PermissionLib.MultiTargetPermission[]
+      memory permissions = new PermissionLib.MultiTargetPermission[](1);
+
+    permissions[0] = PermissionLib.MultiTargetPermission({
+      operation: PermissionLib.Operation.Grant,
+      where: plugin,
+      who: _dao,
+      condition: PermissionLib.NO_CONDITION,
+      permissionId: SimpleStorageBuild2(this.implementation()).STORE_PERMISSION_ID()
+    });
+
+    preparedSetupData.permissions = permissions;
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareUpdate(
+    address _dao,
+    uint16 _currentBuild,
+    SetupPayload calldata _payload
+  )
+    external
+    pure
+    override
+    returns (bytes memory initData, PreparedSetupData memory preparedSetupData)
+  {
+    (_dao, preparedSetupData);
+
+    if (_currentBuild == 0) {
+      address _account = abi.decode(_payload.data, (address));
+      initData = abi.encodeWithSelector(
+        SimpleStorageBuild2.initializeFromBuild1.selector,
+        _account
+      );
+    }
+  }
+
+  /// @inheritdoc IPluginSetup
+  function prepareUninstallation(
+    address _dao,
+    SetupPayload calldata _payload
+  ) external view returns (PermissionLib.MultiTargetPermission[] memory permissions) {
+    permissions = new PermissionLib.MultiTargetPermission[](1);
+
+    permissions[0] = PermissionLib.MultiTargetPermission({
+      operation: PermissionLib.Operation.Revoke,
+      where: _payload.plugin,
+      who: _dao,
+      condition: PermissionLib.NO_CONDITION,
+      permissionId: SimpleStorageBuild2(this.implementation()).STORE_PERMISSION_ID()
+    });
+  }
+
+  /// @inheritdoc IPluginSetup
+  function implementation() external view returns (address) {
+    return simpleStorageImplementation;
+  }
+}
+```
+
+The key thing about to review in this new Plugin Setup contract is its `prepareUpdate()` function. The function only contains a condition checking from which build number the update is transitioning to build 1. Here, it is the build number 0 as this is the only update path we support. Inside, we decode the `address _account` input argument provided with `bytes _date` and pass it to the `initializeFromBuild1` function taking care of intializing the storage that was added in this build.
```

Comment:
Again, `initialize(fromVersion)` and the first ever build being `1`

```suggestion
The key thing to review in this new Plugin Setup contract is its `prepareUpdate()` function. The function only contains a condition checking from which build number the update is transitioning to build `2`. Here, it is the build number `1` as this is the only update path we support. Inside, we decode the `address _account` input argument provided with `bytes _date` and pass it to the `initializeFromBuild1` function taking care of intializing the storage that was added in this build.
```

Reply:
Simple storage plugin update - added to notes! + fixed the wording suggested.